tests for parallel imports and zinject import delay

don-brady opened the pull request:
### Description
This PR supersedes #2

Add `zinject -I <secs> <pool>` injector for testing pool imports with delays.
This replaces the previous method of using dirty ZIL records since we want the delay to occur during the load phase (where the namespace lock is not held and `spa_load_thread` is active).

Also make pool name in `spa_tryimport()` unique.

Added negative and admin tests:
	`zpool_import_parallel_neg`
	`zpool_import_parallel_admin`

grwilson commented on cmd/zinject/zinject.c:
```diff
@@ -896,6 +952,20 @@ main(int argc, char **argv)
 			usage();
 			libzfs_fini(g_zfs);
 			return (0);
+		case 'i':
+			errno = 0;
+			record.zi_timer = (int)strtol(optarg, &end, 10);
+			if (record.zi_timer <= 0 || *end != '\0') {
+				(void) fprintf(stderr, "invalid import delay "
+				    "'%s': must be a positive integer\n",
+				    optarg);
+				usage();
+				libzfs_fini(g_zfs);
+				return (1);
+			}
+			record.zi_timer = SEC2NSEC(record.zi_timer);
+			record.zi_cmd = ZINJECT_DELAY_IMPORT;
+			break;
```
We have a 's' option which sets a duration. Can we combine the code so that we don't duplicate it?

don-brady replied:
How about something like:  `zinject -P import|export -s seconds <pool>` ?

grwilson reviewed (approved):
This looks good. Can you fix the style check failures, then I can merge it.

Commit:
tests for parallel imports with zinject import delay

New delay injector: zinject -P import -s <secs> <pool>
Change test import delays from dirty ZIL to use zinjector
Made pool name in spa_tryimport() unique
Added zpool -a testing
Added test for attempting a pool create during import

Signed-off-by: Don Brady <don.brady@klarasystems.com>

don-brady commented:
Fixed the cstyle errors and squished everything into one commit.

Also note that I finished up  manual testing on real system.  Results look good, an example run result:
```
Starting parallel imports
pool-9089361637256346193: imported in 13 secs
pool-3262361095577314556: imported in 13 secs
pool-12452062456849323294: imported in 16 secs
pool-12255040345250044327: imported in 17 secs
pool-14804618686783290211: imported in 18 secs
pool-1774030350904528497: imported in 18 secs
pool-14579491240965479072: imported in 18 secs
Parallel imports finished in 18 seconds
```
